src/group_theory/order_of_element.lean

```diff
@@ -200,16 +325,80 @@ calc ∑ m in (finset.range n.succ).filter (∣ n), (finset.univ.filter (λ a :
     λ h, ⟨order_of_le_of_pow_eq_one hn h, order_of_dvd_of_pow_eq_one h⟩⟩
 end))
 
-section
-local attribute [instance] set_fintype
+
+
+end finite_monoid
+
+section finite_cancel_monoid
+variables {α} [fintype α] [decidable_eq α] [left_cancel_monoid α]
+
+lemma exists_pow_eq_one (a : α) : ∃i > 0, a ^ i = 1 :=
```

Comment:
`∃ n, 0 < n ∧ a ^ n = 1` and `∃i > 0, a ^ i = 1` are different to Lean, so I think it'd make sense for this lemma and the definition of `order_of` to match.